src/components/Inputs/BaseInput/BaseInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+interface BaseInputProps {
+  type: 'text';
```

Comment:
I would add an id or name to use those properties. Think of this input for any possible form.